### DISCUSSION TITLE
Turborepo Tailwind CSS starter - Failed to compile error

### DISCUSSION THREAD

**Discussion by rogix** [vercel/turborepo#3889] [Help] [answered]
**Problem:**

When running the Tailwind example that uses `tsup` to bundle the package, an error occurs when the Button component is saved, as shown below:

```bash
error - ../../node_modules/.pnpm/next@13.1.6_m5sxuueb27gk6ddc5gums6vtgq/node_modules/next/dist/build/webpack/loaders/css-loader/src/index.js??ruleSet[1].rules[3].oneOf[7].use[1]!../../node_modules/.pnpm/next@13.1.6_m5sxuueb27gk6ddc5gums6vtgq/node_modules/next/dist/build/webpack/loaders/postcss-loader/src/index.js??ruleSet[1].rules[3].oneOf[7].use[2]!../../packages/ui/dist/index.css

Error: ENOENT: no such file or directory, open '.../packages/ui/dist/index.css'
```

Additionally, an error message briefly flashes on the screen, which is distracting.

**Solution:**

One way to fix this problem is to remove the `clean` option from the `tsup.config.ts` file:

```js
export default defineConfig((options: Options) => ({
  treeshake: true,
  splitting: true,
  entry: ['src/**/*.tsx'],
  format: ['esm'],
  dts: true,
  minify: true,
  // clean: true, <-- remove this
  external: ['react'],
  ...options,
}))
```

However, there might be a better solution that avoids having to remove the `clean` option. Unfortunately, this is not documented in the Tailwind documentation.

Please let me know if you have any insights or suggestions on how to fix this issue more elegantly.

**Comment by rogix** [comment, marked as answer]
I've found a helpful solution for anyone who needs it. 

Instead of using the `clean` option in the `tsup.config.ts`, you can add it to the `package.json` file of the library, for example, in the `packages/ui` folder. You can achieve this by adding the following code to the scripts section of the `package.json` file:

```js
  "scripts": {
    "build": "pnpm run clean && tsup",
    "clean": "rm -rf dist", // This line removes the dist folder
    "dev": "tsup --watch",
    "check-types": "tsc --noEmit"
  },
```

This approach gives you the same benefits as the `tsup` clean option without encountering any errors.

I hope this helps!

**Reply by anthonyshew** [reply]
We have updated the Tailwind example recently so this workaround shouldn't be necessary anymore!
